### PR TITLE
This fixes an issue in our html sanitization when no IHyperlinkFormatter is registered

### PR DESCRIPTION
**Pull request by cutz**
The old implementation resulted in character data being dropped when no IHyperlinkFormatter was found in the registry. This resulted in the added test producing a value of `<html><body><p></p></body></html>` rather than the expected value `<html><body><p>look at this</p></body></html>`

### CODE REVIEW

**Comment by jamadden** [comment]
How does this interact with #40? I'd really prefer to get that merged.

**Comment by cutz** [comment]
Looking at the changes in #40 the changes to test_html I made are probably going to need to be manually rebased, but the fix in `_find_links_in_text` shouldn't be affected.

**Comment by cutz** [comment]
This fix is for an issue I'm seeing in some tests in another area where I don't have an IHyperlinkFormatter registered. That said, in all deployed cases we should have that, as this package registers one. I can work around it by registering an IHyperlinkFormatter in the tests, but given that we are [querying the utility](https://github.com/NextThought/nti.contentfragments/blob/fix-no-link-formatter/src/nti/contentfragments/html.py#L120) it seemed like we would want this to work without on. That gets me unblocked if we need to hold on this until #40 is taken care of. I suppose another solution would be to change that to `getUtility`. I'm not sure of the original intent in terms of if that is optional or not.

File: src/nti/contentfragments/html.py
Comment (jamadden) [inline src/nti/contentfragments/html.py]: Instead of introducing a new shortcut code path, given that there's already a conditional immediately following this, that, when it fails, just yields the token (exactly what the new condition is doing), I would probably have changed the constructor: `self.link_finder = component.queryUtility(IHyperlinkFormatter, dummy_formatter)`, where  `dummy_formatter` looks like:
```python
class _DummyFormatter(object):
    def find_links(self, text):
       # Need to return a single element; this is interpreted as not actually finding links
        return (text,)
dummy_formatter = _DummyFormatter()
```
(Choosing better names, of course.)

That also removes the condition from the original line (`if self.link_finder else ()`) because we can always assume `link_finder` is non-None.

An even simpler change would be `if self.link_finder else (text,)`, but that still leaves the condition.

**Comment by jamadden** [comment]
I think #40 is ready, it just needs somebody to click 'approve'. I would ask (or beg!) you to apply the changes from this PR on top of the changes from that one, because they're much more extensive.

**Comment by cutz** [comment]
Rebased on top of `issue39` branch. We can wait to merge this until after #40

**Comment by jamadden** [comment]
Thanks!

(We need a `# pragma: no cover` on the `raise NotImplemented` line to get coverage back to 100% and make the PR green.)

**Review by jamadden — APPROVED** [review]
Works for me.

**Comment by cutz** [comment]
What's the best merge approach? Merge and close out #40 and rebase this on master, or close #40 and merge this branch as is?

**Comment by jamadden** [comment]
Since you did a clean rebase, we can merge #40 and then this PR should automatically go back to just showing its unique commits; no need to rebase again. Alternatively, merging this PR would automatically close #40 because it has the same commits. 

I'd prefer to go with the first approach, just because I like clicking the merge button :)

**Comment by jamadden** [comment]
Hmm. So maybe not, for whatever reason. Rebasing on master would make the cleanest history then.

**Comment by jamadden** [comment]
1.8.0 is on PyPI.

**Pull request merged**
